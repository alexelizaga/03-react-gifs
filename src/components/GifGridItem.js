import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url}) => {
  return (
    <div data-testid='GifGridItem' className='card animate__animated animate__fadeIn'>
      <img data-testid='GifGridItemImg' src={url} alt={title} />
      <p data-testid='GifGridItemTitle'>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}