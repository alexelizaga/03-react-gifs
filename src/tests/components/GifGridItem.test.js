import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import { GifGridItem } from '../../components/GifGridItem';

describe('Test <GifGridItem />', () => {

  const props = {
    title: 'title',
    url: 'https://www.example.com'
  }

  test('should render <GifGridItem /> correctly', () => {
    const tree = renderer
      .create(<GifGridItem {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have a parragraph with the title', () => {
    render(<GifGridItem {...props} />);

    expect(screen.getByTestId('GifGridItemTitle').textContent.trim()).toBe(props.title);
  });

  test('should have img with src and alt', () => {
    render(<GifGridItem {...props} />);

    const img = screen.getByTestId('GifGridItemImg');
    expect(img).toBeInTheDocument();
    expect(img.getAttribute('src')).toBe(props.url);
    expect(img.getAttribute('alt')).toBe(props.title);
  });

  test('should have animate__fadeIn', () => {
    render(<GifGridItem {...props} />);

    const div = screen.getByTestId('GifGridItem');
    expect(div.className.includes('animate__fadeIn')).toBeTruthy();
  });

});