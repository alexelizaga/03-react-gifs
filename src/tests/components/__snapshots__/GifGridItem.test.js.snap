// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> should render <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
  data-testid="GifGridItem"
>
  <img
    alt="title"
    data-testid="GifGridItemImg"
    src="https://www.example.com"
  />
  <p
    data-testid="GifGridItemTitle"
  >
    title
  </p>
</div>
`;
